library/Icinga/Crypt/AesCrypt.php

```diff
@@ -50,22 +60,27 @@ class AesCrypt
     /** @var string The authentication tag which is passed by reference when using AEAD cipher mode */
     private $tag;
 
-    /** @var string The cipher method */
-    private $method = 'AES-128-GCM';
+    private $method;
 
     const GCM_SUPPORT_VERSION = '7.1';
 
-    public function __construct($random_bytes_len = 128)
+    public function __construct()
     {
-        if (version_compare(PHP_VERSION, self::GCM_SUPPORT_VERSION, '<')) {
-            $this->method = 'AES-128-CBC';
+        foreach (self::METHODS as $key => $method) {
+            if ($key === 0 && version_compare(PHP_VERSION, self::GCM_SUPPORT_VERSION, '<')) {
+                continue;
+            }
+            if (in_array($method, openssl_get_cipher_methods())) {
+                $this->method = $method;
+                break;
+            }
         }
```

Comment:
Please put that into the getter, and only detect it in case there's no method set yet.

Also:
* Drop the `$key === 0` and php version check, that's redundant
* Cache the result of `openssl_get_cipher_methods()` in a variable